experiment/measurer/measure_manager.py

```diff
@@ -494,24 +495,84 @@ def extract_corpus(self, corpus_archive_path) -> bool:
         extract_corpus(corpus_archive_path, unit_blacklist, self.corpus_dir)
         return True
 
-    def archive_crashes(self, cycle):
-        """Archive this cycle's crashes into filestore."""
-        if not os.listdir(self.crashes_dir):
-            logs.info('No crashes found for cycle %d.', cycle)
+    def save_crash_metadata(self, crashes, cycle):
+        """Stores crash metadata in database and per-cycle metadata file."""
+        if not crashes:
             return
 
-        logs.info('Archiving crashes for cycle %d.', cycle)
+        logs.info('Storing crashes in database.')
+        experiment_name = experiment_utils.get_experiment_name()
+        crash_time = experiment_utils.get_cycle_time(cycle)
+        for crash_key in crashes:
+            crash = crashes[crash_key]
+            if db_utils.query(models.Crash).filter_by(
+                    experiment=experiment_name,
+                    fuzzer=self.fuzzer,
+                    benchmark=self.benchmark,
+                    crash_key=crash_key).first():
+                logs.info('Crash already exists in database: %s', crash)
+                continue
+
+            logs.info('New crash found, storing in database: %s', crash)
+            try:
+                db_utils.add_all([
+                    models.Crash(experiment=experiment_name,
+                                 fuzzer=self.fuzzer,
+                                 benchmark=self.benchmark,
+                                 crash_key=crash_key,
+                                 crash_testcase=crash.crash_testcase,
+                                 crash_type=crash.crash_type,
+                                 crash_address=crash.crash_address,
+                                 crash_state=crash.crash_state,
+                                 crash_stacktrace=crash.crash_stacktrace,
+                                 crash_time=crash_time)
+                ])
+            except Exception:  # pylint: disable=broad-except
+                # Can happen due to race condition with db updates.
+                logs.error('Failed to store crash.')
+
+        logs.info('Storing crashes in metadata file.')
+        trial_crashes_dir = posixpath.join(self.trial_dir, 'crashes')
```

Comment:
Why do we need a crash metadata file if we are storing stuff in the db?

Reply:
Good for debugging or if someone wants to look at what crashes were found in every cycle. 
Also, we need to expose crash db data in data.csv.gz, i will do that in another cl.

Reply:
Ha I think this is a good reason *not* to have this data. If it does in data.csv.gz it should come from the db and not from some other source.
Or do you need to do this because we want the crash data in data.csv.gz to be per cycle and what you did here doesn't support that?
Frankly, I'm having second thoughts about the model chosen here. We store everything as a snapshot. This is changing that and adding complexity.

Reply:
Removed